### ISSUE TITLE
staticcheck: same exit condition checked twice

### ISSUE THREAD

**Issue by lu4p** [dominikh/go-tools#890]
I found this in my code, note `err` is checked twice without a new value being assigned to `err`:
```go
func connect() (net.Conn, error) {
	conn, err := net.Dial("tcp", s.addr)
	if err != nil {
		return nil, err
	}
	caPool := x509.NewCertPool()
	caPool.AddCert(s.cert)

	config := tls.Config{RootCAs: caPool, ServerName: s.domain}
	tlsconn := tls.Client(conn, &config)
	if err != nil {
		return nil, err
	}
	return tlsconn, nil
}
```

I would suggest adding a check for duplicated exit conditions (checking a value and then returning), with unchanged values.
This could also  be restricted to errors only, I don't have a strong opinion here.

Message would be something like:
`Ineffective exit condition, same condition exited earlier and  none of the values changed`

Suggested fix:
`Change a value in this condition, or remove this block`

What counts as a potential change:
- mutated the value of a variable (assigning, incrementing etc.)
- variable has been shadowed
- variable was passed as a pointer to a function

(or maybe between the first and the second condition the variable was referenced, probably simpler and faster)

This check should exclude:
- global variables
- pointers passed as parameters to the function 
- goroutines, which reference a variable used in the condition, and are created before the second condition

basically all cases, where another goroutine could mutate the value of a variable between the two conditions.

**Comment by ainar-g** [comment]
I think [`nilness`] detects your particular example.  The message would be something like:

```none
file.go:123:45: impossible condition: nil != nil
```

[`nilness`]: https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/nilness/cmd/nilness
